src/enum_flags.h

```diff
@@ -37,53 +39,68 @@ template<typename Enum> class flags
 {
 public:
    using enum_t = Enum;
+   using int_t  = typename std::underlying_type<enum_t>::type;
+
+   template<typename T> using integral =
+      typename std::enable_if<std::is_integral<T>::value, bool>::type;
 
    inline flags() = default;
    inline flags(Enum flag)
-      : m_i{static_cast<int>(flag)} {}
+      : m_i{static_cast<int_t>(flag)} {}
```

Comment:
Why `int_t`, I'd think that a more general naming like `num_t` would be more appropriate here.

Reply:
Yeah, in this case "int" just means "integer"/"integral", no particular relation (necessarily) to the built-in `int` type. A "number" could be a real. I have a slight leaning to leaving it as-is, but if you have a strong objection/preference...?